src/Symfony/Component/Notifier/Bridge/Ntfy/NtfyTransport.php

```diff
@@ -87,7 +87,7 @@ protected function doSend(MessageInterface $message): SentMessage
         $headers = [];
 
         if (null !== $this->user && null !== $this->password) {
-            $headers['Authorization'] = 'Basic '.rtrim(base64_encode($this->user.':'.$this->password), '=');
+            $headers['Authorization'] = 'Basic '.base64_encode($this->user.':'.$this->password);
```

Comment:
@mikaelkael Would be great if you could have a look here as you contributed the initial code in #50131.

Reply:
> @xabbuh: as explain here https://docs.ntfy.sh/publish/#username-password, you don't have final sign equal and this modification breaks associated test https://github.com/symfony/symfony/actions/runs/10555314977/job/29238683518#step:8:1929. As all tests use mocks without direct access to a ntfy server, perhaps your modification is possible but you need to also change tests to be compliant

Suggested page not explain you don't have final equal sign. But in case you encode `testuser:fakepassword` string from Ntfy docs base64 output is without equal sign(s).

Doing tests is totally out of my scope. I just only wanted report bug I have found. Somebody asks me for PR, I did ... sorry :-(

Reply:
The fix looks correct to me, as Base64 encoded strings do not always end with an = character. The = character is used as padding to ensure that the length of the encoded string is a multiple of 4 characters, which is required by the Base64 encoding standard.

So yes, the tests should be adjusted

https://3v4l.org/BHa9R

Reply:
@xabbuh: you have the mention of no equal sign in https://docs.ntfy.sh/publish/#query-param, specifically for query param and as @OskarStark said, the case of the doc 'testuser:fakepassword' doesn't have need padding. I will try to reproduce https://github.com/symfony/symfony/issues/58085 with a real server (not mocked)

Reply:
@jirikmik Did you encounter any real issue when `=` characters where stripped?

Reply:
Isn't this plain HTTP basic auth? Looking at [the specification](https://datatracker.ietf.org/doc/html/rfc7617#section-2), I don't see why we should strip the trailing `=` characters. In fact, the example in the spec does not strip them.

Additionally, the HTTP client supports basic and bearer auth. I think, we should not assemble that header ourselves if the HTTP client can do that already for us.

https://github.com/symfony/symfony/blob/8d966084665992b607284144229ec5e80f589f7f/src/Symfony/Contracts/HttpClient/HttpClientInterface.php#L27-L30

Reply:
> @jirikmik Did you encounter any real issue when `=` characters where stripped?

@xabbuh yes, #58085

Reply:
@derrabus Good point. Yes, it seems as plain HTTP basic auth. I also looked this package has http client as dependency, so it can use it for this ...

Reply:
> HTTP client would not strip any = sign, so I agree it should be done by http client, but if we need to trim according to their docs, we have no choice

@OskarStark There is nothing about stripping = sign in their docs.

Reply:
Indeed, I didn't check the link from @mikaelkael, and thought it is present, but there is no mention of "no equal sign", so http client should handle that.
Ofc a check with a real server would be appreciated

Reply:
Ah it is:
![image](https://github.com/user-attachments/assets/64c8c9c0-10bd-4ad2-ac80-d29e6bbdc16c)

Reply:
I think screenshot from Ntfy.sh docs is related to "query param" auth, not Basic auth ...

Reply:
In none of the example in the doc, we can find padding equal (for any langage, not only PHP) but it's not a valid reason. During my dev, it didn't made problem. There is perhaps a change with recent version of ntfy. The only way is to have a reproductible test case. As I know in symfony, we didn't depend on external service in test. So I will try to write a failing test case in my https://github.com/mikaelkael/ntfy-notifier (the repository on which I work before integration in symfony).

Reply:
If I read the linked documentation correctly, stripping the trailing `=` characters is documented for authenticating via the query parameter,but not for basic auth. This means the proposed change is correct.

Reply:
I was able to reproduce the bug https://github.com/mikaelkael/ntfy-notifier/commit/a9e19dd6c0e2397968b51dab7cb650cb20718ba6, I confirm that the correction is good and the only thing to change in test is: https://github.com/mikaelkael/ntfy-notifier/commit/a9e19dd6c0e2397968b51dab7cb650cb20718ba6#diff-cfb9cdd160d285443c2bf73a6307b616ce74d3431a19d887ad290425606d2471L114